packages/api/src/server/application/service/application.factory.ts

```diff
@@ -396,17 +401,39 @@ export class ApplicationFactory {
   }
 
   // @internal It'll check that the repository exists or not
-  async validateGitProps(gitRequestDTO: GitRequestDTO) {
-    const gitUrl = `${gitRequestDTO.repoUrl}/tree/${gitRequestDTO.gitRef}/${gitRequestDTO.contextDir}`;
-    try {
-      const response = await this.httpService.axiosRef.head(gitUrl);
-      if (response.status === 200) return true;
-    } catch (error) {
-      this.logger.error('GitSource', error);
+  async validateGitProps(repoUrl: string, gitRef: string, contextDir: string) {
+    const gitUrl = this.generateGitUrl(repoUrl, gitRef, contextDir);
+    if (gitUrl.startsWith('https://gitlab')) {
+      try {
+        const response = await this.httpService.axiosRef.get(gitUrl);
+        // @internal for the valid repository Gitlab sends the list of valid urls
+        const regex = /\bhttps?:\/\/[^\s,"}]+\b/g;
+        const gitlabSource = response.data.match(regex).includes(gitUrl.replace(/\/$/, ''));
```

Comment:
I think it will be more expressive if in the comment you write

> this regular expression matches any valid URL that starts with 'http://' or 'https://'
Instead of 
> "for the valid repository Gitlab sends the list of valid urls"
I guess that's what this regex is doing, right?
even better if you rename the name of the variable from `regex` to `validUrlPattern` then you don't have to use comments. This was just one example and true in every other place where you have used the regex pattern